common/common.h

```diff
@@ -3,7 +3,9 @@
 #pragma once
 
 #include "llama-cpp.h"
+#include "chat-template.hpp"
```

Comment:
Including this here will pollute a lot of source files with a `json.hpp` dependency. We should avoid this.

Reply:
Good point thanks, now forward declared.